src/lib/models/inorder/Core.cc

```diff
@@ -34,16 +33,37 @@ Core::Core(MemoryInterface& instructionMemory, MemoryInterface& dataMemory,
           [this](auto instruction) { storeData(instruction); },
           [this](auto instruction) { raiseException(instruction); },
           branchPredictor, false),
-      writebackUnit_(completionSlots_, registerFileSet_, [](auto insnId) {}) {
-  // Query and apply initial state
-  auto state = isa.getInitialState();
-  applyStateChange(state);
-};
+      writebackUnit_(completionSlots_, registerFileSet_, [](auto insnId) {}){};
 
 void Core::tick() {
   ticks_++;
+  isa_.updateSystemTimerRegisters(&registerFileSet_, ticks_);
+
+  switch (status_) {
+    case CoreStatus::idle:
+      idle_ticks_++;
+      return;
+    case CoreStatus::switching: {
+      if (fetchToDecodeBuffer_.isEmpty() && decodeToExecuteBuffer_.isEmpty() &&
```

Comment:
Ensure the pipeline is empty and there's no active exception before context switching.